app/models/consent/implied.rb

```diff
@@ -0,0 +1,76 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Consent::Implied
+  def initialize(client:)
+    @client = client
+  end
+
+  private def current_consent_type
+    consent_type = no_release_string
+    consent_type = full_consent_string if @client.active_consent_form&.consent_type == full_release_string
+    consent_type = revoked_consent_string if @client.newest_consent_form&.revoked?
+    consent_type
+  end
+
+  def no_release_string
+    'Implied Consent'
+  end
+
+  def full_consent_string
+    'Expanded Consent'
+  end
+
+  def revoked_consent_string
+    'Consent Revoked'
+  end
+
+  def partial_release_string
+    'Implied Consent'
+  end
+
+  def full_release_string
+    'Expanded Consent'
+  end
+
+  def release_current_status
+    # TODO: COMPLETE THIS
+    consent_text = @client.class.no_release_string
+    consent_text = @client.class.full_consent_string if current_consent_type == full_release_string
+    consent_text = @client.class.revoked_consent_string if current_consent_type == revoked_consent_string
```

Comment:
This _looks_ right to me.  Maybe no longer a TODO?

Reply:
Oh, except, I _think_ you want to use the local methods instead of reaching out to the client:
```suggestion
    consent_text = no_release_string
    consent_text = full_consent_string if current_consent_type == full_release_string
    consent_text = revoked_consent_string if current_consent_type == revoked_consent_string
```

Reply:
yes, the TODO should be removed, and we should use the local methods here. I missed the change after adding the local methods.